### PR TITLE
fix(mod_perl2.sls): fix a2enmod perl2 error

### PR DESCRIPTION
**Pull request by SynPrime**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

If I run the state `apache.mod_perl2` on debian, the state will fail with error "Module perl2 does not exist".
The name of the module is just "perl".

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by myii** [comment]
Definitely appears to be a problem.

https://travis-ci.org/myii/apache-formula/jobs/607230685#L1935-L1948

```sls
                 ID: a2enmod perl2
           Function: cmd.run
             Result: False
            Comment: Command "a2enmod perl2" run
            Started: 16:43:24.041752
           Duration: 74.183 ms
            Changes:   
              ----------
              pid:
                  1523
              retcode:
                  1
              stderr:
                  ERROR: Module perl2 does not exist!
```

The solution may be better to introduce via. the map, though.

**Comment by myii** [comment]
This also appears to be a good fix across all of the Debian-based images:

* https://travis-ci.org/myii/apache-formula/builds/607305170

FreeBSD still needs to be tested, though.

File: apache/mod_perl2.sls [hunk 1]
```diff
@@ -17,9 +17,9 @@ mod-perl2:
       - service: apache
 
 {% if grains['os_family']=="Debian" %}
-a2enmod perl2:
+a2enmod perl:
```
Comment (myii) [inline apache/mod_perl2.sls]: Let's not change the state identifier unnecessarily:

```suggestion
a2enmod perl2
```

Comment (SynPrime) [reply apache/mod_perl2.sls]: @myii If I reset the identifier, the state will fail again. So we need the `name` argument like that:

```
a2enmod perl2:
  cmd.run:
    - name: a2enmod perl
    - unless: ls /etc/apache2/mods-enabled/perl.load
    - order: 225
    - require:
      - pkg: mod-perl2
    - watch_in:
      - module: apache-restart
    - require_in:
      - module: apache-restart
      - module: apache-reload
      - service: apache
```
Am I right?

Comment (myii) [reply apache/mod_perl2.sls]: @SynPrime You're absolutely right, when I ran the tests, I didn't change the state identifier.  In this case, it's being used implicitly by `cmd.run`, so it _is_ required to be changed in this case!  Let's consider this resolved and merge this PR.  Thanks again.

**Pull request merged**